compiler/src/dotty/tools/io/ZipArchive.scala

```diff
@@ -160,7 +160,7 @@ final class FileZipArchive(jpath: JPath, release: Option[String]) extends ZipArc
     override def sizeOption: Option[Int] = Some(zipEntry.getSize.toInt)
   }
 
-  @volatile lazy val (root, allDirs): (DirEntry, collection.Map[String, DirEntry]) = {
+  lazy val (root, allDirs): (DirEntry, collection.Map[String, DirEntry]) = {
```

Comment:
This should be allowed and should not change. Volatile should also work on `lazy val` because the annotation applies in the `private var` that is used to encode the `lazy val`.

Reply:
Aren't `lazy vals` now thread-safe by default?

Reply:
Right